src/woo_publications/publications/forms.py

```diff
@@ -47,3 +50,88 @@ def clean(self):
                 self.add_error("identifier", error=_("This field is required."))
 
         return cleaned_data
+
+
+class PublicationStatusForm(forms.ModelForm):
+    def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop("request", None)
+        super().__init__(*args, **kwargs)
+
+        self.initial_publicatiestatus = self.instance.publicatiestatus or ""
+
+        if self.instance and self.fields.get("publicatiestatus"):
+            allowed_values: list[str] = [
+                status.target.value
+                for status in self.instance.get_available_publicatiestatus_transitions()
+            ]
+
+            if selected_publicatiestatus := self.instance.publicatiestatus:
+                allowed_values.append(selected_publicatiestatus)
+
+            self.fields["publicatiestatus"].choices = [  # pyright: ignore[reportAttributeAccessIssue]
+                choice
+                for choice in PublicationStatusOptions.choices
+                if choice[0] in allowed_values
+            ]
+
+
+class PublicationAdminForm(PublicationStatusForm):
+    def save(self, commit=True):
+        assert is_authenticated_request(self.request)
+        publicatie_status = self.cleaned_data["publicatiestatus"]
+
+        publication = super().save(commit=commit)
+        # cannot force the commit as True because of the AttributeError:
+        # 'PublicationForm' object has no attribute 'save_m2m'.
+        # So we manually save it after running the super
+        publication.save()
+
+        self.instance.publicatiestatus = self.initial_publicatiestatus
+
+        if publicatie_status == PublicationStatusOptions.published:
+            publication.publish(self.request, user=self.request.user)
```

Comment:
Can split this up depending on actual status changes but this seems to be the more practical solution code wise.